lib/resque/worker.rb

```diff
@@ -228,7 +228,7 @@ def work(interval = 5.0, &block)
           break if interval.zero?
           log_with_severity :debug, "Sleeping for #{interval} seconds"
           procline paused? ? "Paused" : "Waiting for #{queues.join(',')}"
-          sleep interval
+          @signal_handler.handle_signal(interval)
```

Comment:
Hmm, if we move this out of the unless, waiting interval is issued even if we run a job (behavior changes).
I am not getting good idea yet. I will think of how to resolve this. Any idea is welcome.